### ISSUE TITLE
Sprite's keyframes deletion by some actions in addon's UI

### ISSUE THREAD

**Issue by EvgeneKuklin** [Aodaruma/coa_tools2#34]
**Describe the bug**
Some actions in the addon's UI deletes animation keyframes of sprite objects

**Video**
[YouTube video](https://youtu.be/YcRJKmtTGmU)

**To Reproduce**
1. Open clean scene
2. Open the addon's "N" menu
3. Click "Create new sprite object"
4. Click "Re / Import sprites" and import some sprites
5. In the addon's block "Cutout Animations" click plus button and add new animation collection
6. In the addon's block "Object properties" click on any imported spite (.png usually)
7. Press I on keyboard to insert a keyframe and choose location, for example
8. In the addon's block "Cutout Animations" click on "Restpose" or "NO ACTION" and the on the ani,ation collection (NewCollection by default)
9. Now we lost all sprite's keyframes

**Expected behavior**
I expect that sprite's keyframes will be on their places

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Blender 3.5/3.6
 - CoaTools2's commit: (Master) 51bfbc52b5867f7d192659cbd935212e79420870

**Additional context**
I think the bug is critical: not only it's not allowing to use direct animation of sprites but also could delete person's hard work without notice

**Comment by Aodaruma** [comment]
confirmed, it seems to be related to `animation_handly.py`.
I found COATOOLS2_OT_AddKeyframe at [L46-L207 in `animation_handly.py`](https://github.com/Aodaruma/coa_tools2/blob/51bfbc52b5867f7d192659cbd935212e79420870/coa_tools2/operators/animation_handling.py#L46-L207) and it looks to be used in [Object Property of coa_tools2](https://github.com/Aodaruma/coa_tools2/blob/51bfbc52b5867f7d192659cbd935212e79420870/coa_tools2/ui.py#L329-L358).

using "add keyframe" button, it stored correctly in the animation collection whenever change anim_collection_index.
![image](https://github.com/Aodaruma/coa_tools2/assets/15927814/d0faf963-4499-43d0-ad53-e7a668d8b1cd)

therefore it is likely in the specifications of coa_tools, but i think it is not user-friendly feature for being lost all keyframes when we edit animation collections

**Comment by EvgeneKuklin** [comment]
I think there might be a related problem: Animation lenght is also resets to default 250 frames in similar scenarios except when we set it in coa's ui

**Comment by Aodaruma** [comment]
yeah. Not only this problem, but we have seen some problems with this addon where data edited without going through operators functionality is not saved.

Anyway, possible solution to this issue is: 
1. Automatically register keyframes for objects and bones below the sprite object when animation collections are selected
2. When creating the first animation collections, keyframes below the sprite object should be registered in the first animation collections, and the animation length should be applied to the originally registered end frame (this may be rather deprecated; it would be more appropriate to indicate a modal which shows "all keyframes and animation data will be lost")

**Comment by Aodaruma** [comment]
@EvgeneKuklin 
Sorry for the lack of response. It should be fixed in this release.
Please check if it has been properly resolved when you have some time :)
https://github.com/Aodaruma/coa_tools2/releases/tag/v2.3.0
